### PR TITLE
Random Battles: 2023-04 End-of-month changes

### PR DESCRIPTION
**Pull request by adrivrie**
April monthly randbats balance. Ideally is merged as soon as possible so the May stats are based on the new levels.

### CODE REVIEW

**Review by livid-washed — APPROVED** [review]

File: data/mods/gen3/random-teams.ts [hunk 1]
```diff
@@ -246,6 +246,8 @@ export class RandomGen3Teams extends RandomGen4Teams {
 		// First, the high-priority items
 		if (species.name === 'Ditto') return this.sample(['Metal Powder', 'Quick Claw']);
 		if (species.name === 'Farfetch\u2019d') return 'Stick';
+		if (species.name === 'Latias') return 'Soul Dew';
+		if (species.name === 'Latios') return 'Soul Dew';
```
Comment (ACakeWearingAHat) [inline data/mods/gen3/random-teams.ts:250]: these can be combined with a ||

Comment (ACakeWearingAHat) [reply data/mods/gen3/random-teams.ts:250]: same for every gen

**Review by ACakeWearingAHat — CHANGES_REQUESTED** [review]

**Comment by adrivrie** [comment]
This PR now also includes a more general Random Battles patch: adding soul dew to (non-mega) Lati@s and revamping their sets in gens 3-6, and removing Toxtricity Low-Key from gen9 randbats.

This was combined into the same PR to save some merging effort as these changes would likely have caused a merge conflict otherwise. If this is inconvenient or (too) bad practice, please let us know the preferred course of action

File: data/mods/gen3/random-teams.ts (hunk 1 above)
Comment (adrivrie) [reply data/mods/gen3/random-teams.ts:250]: Eh, sure

**Review by ACakeWearingAHat — APPROVED** [review]

**Pull request merged**